src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -423,14 +425,14 @@ private void CancellationCallback()
                 {
                     // Remove all watches (inotiy_rm_watch) and clear out the map.
                     // No additional watches will be added after this point.
-                    SysCall(fd => {
-                        foreach (int wd in _wdToPathMap.Keys)
+                    SysCall((fd, runningInstance, _) => {
```

Comment:
The line numbers seem to have changed since my original comment. What I commented on was watcher being typoed as wacther. Would have left a line comment but had issues getting it to work on IE for RT